lib/jormungandr/src/Cardano/Wallet/Jormungandr/Binary.hs

```diff
@@ -372,6 +393,47 @@ putTx inputs outputs = do
         putAddress address
         putWord64be $ getCoin coin
 
+{-------------------------------------------------------------------------------
+                                 Stake Distribution
+-------------------------------------------------------------------------------}
+
+data StakeDistribution = StakeDistribution
+    { epoch :: Word64
+    , stake :: StakePools
+    } deriving (Show, Eq, Generic)
+
+-- | Coins are stored as Lovelace (reminder: 1 Lovelace = 1e-6 ADA)
+newtype Stake = Stake { getStake :: Word64 } deriving (Show, Eq)
+
+data StakePools = StakePools
+    { dangling :: Word64
+    , pools :: [(PoolId,Stake)]
+    , unassigned :: Word64
+    } deriving (Eq, Show)
+
+instance FromJSON StakeDistribution where
+    parseJSON = genericParseJSON defaultOptions
+
+instance FromJSON StakePools where
+    parseJSON = withObject "StakePools" $ \sp -> do
+        d <- sp .: "dangling"
+        p <- sp .: "pools"
+        a <- sp .: "unassigned"
+        let poolParser =  withArray "pairArrays" $ \arr ->
+                case V.toList arr of
+                    [(String txt),(Number st)] ->
+                        case fromText txt of
+                            Left (TextDecodingError err) -> fail err
+                            Right bs ->
+                                case toBoundedInteger st of
+                                    Nothing ->
+                                        fail "expected stake to be integer"
+                                    Just stake ->
+                                        return (bs, Stake stake)
+                    _ -> fail "expected poolId and stake array in pool's array"
+        p' <- mapM poolParser p
+        pure $ StakePools d p' a
```

Comment:
Whola! What about using generics here :blush: ?

Reply:
yes! Proper `FromJson` of `Stake` and `PoolId` and generics can be used. Moreover, the errors msg are more precise than in my hand-crafted version :-)